ONSdigital/dis-authorisation-client-js
# Feature/core renewal functionality

### What

[This](https://jira.ons.gov.uk/browse/DIS-2084) adds the core functionality for client side JS auth.

**Notes**

- A single instance of SessionManagement is used throughout.
- Removed all EnableNewSignIn conditionals
- Storage, popouts, and warnings are handled outside library
- Uses 4 callbacks:
  - onRenewSuccess: if session renews successfully, used to call `startRefeshAndSession`
  - onRenewFailure: if session renewal is unsuccessful, used to throw an error
  - if checkSessionOnInit true during init (used tot see if returning from outside florence
    - onSessionValid
    - onSessionInvalid
- Removes a logout user call which was in `endManualSession` before but that has also been removed

If you have any issues getting it setup, give me a message on slack.

### How to review
I've left console logs in place so it's easier to see what is happening.

You'll need [this](https://github.com/ONSdigital/florence/pull/1083) test branch of florence.

Setup

- Pull this branch
- Open terminal at root library
- Use `npm link`
- Open terminal at root of florence library
- Use `npm link dis-authorisation-client-js` (this will allow the local version of this library to be used)
- Run florence
  - port forward to the api router - `dp ssh sandbox publishing 2 -p 23200:localhost:10800`
  - I was running the home publishing stack
  - `make dev` in florence

Open chrome console and you should be able to see the general flow of what is happening.
![image](https://github.com/user-attachments/assets/8843fa63-d97b-4b46-8bc0-375dcd662524)

You can uncomment lines 78, 80, and 81 in `SignIn.jsx` within the florence branch to reduce the initial expiry times.


### Who can review

Not me
